numpy/core/src/multiarray/array_coercion.c

```diff
@@ -663,7 +667,7 @@ handle_scalar(
     if (descr == NULL) {
         return -1;
     }
-    if (handle_promotion(out_descr, descr, flags) < 0) {
+    if (handle_promotion(out_descr, descr, fixed_DType, flags) < 0) {
```

Comment:
Here too it seems the documentation does not describe the parameters correctly, so it is difficult to follow the logic.